src/tile.h

```diff
@@ -188,64 +198,65 @@ class Tile : public Cylinder
+	                     std::shared_ptr<Creature> actor = nullptr) const override;
+	ReturnValue queryMaxCount(int32_t index, std::shared_ptr<const Thing> thing, uint32_t count,
+	                          uint32_t& maxQueryCount, uint32_t flags) const override final;
+	ReturnValue queryRemove(std::shared_ptr<const Thing> thing, uint32_t count, uint32_t flags,
+	                        std::shared_ptr<Creature> actor = nullptr) const override;
+	std::shared_ptr<Thing> queryDestination(int32_t& index, std::shared_ptr<const Thing> thing,
+	                                        std::shared_ptr<Item>& destItem, uint32_t& flags) override;
 
-	void addThing(Thing* thing) override final;
-	void addThing(int32_t index, Thing* thing) override;
+	void addThing(std::shared_ptr<Thing> thing) override final;
+	void addThing(int32_t index, std::shared_ptr<Thing> thing) override;
 
-	void updateThing(Thing* thing, uint16_t itemId, uint32_t count) override final;
-	void replaceThing(uint32_t index, Thing* thing) override final;
+	void updateThing(std::shared_ptr<Thing> thing, uint16_t itemId, uint32_t count) override final;
+	void replaceThing(uint32_t index, std::shared_ptr<Thing> thing) override final;
 
-	void removeThing(Thing* thing, uint32_t count) override final;
+	void removeThing(std::shared_ptr<Thing> thing, uint32_t count) override final;
 
-	bool hasCreature(Creature* creature) const;
-	void removeCreature(Creature* creature);
+	bool hasCreature(std::shared_ptr<Creature> creature) const;
+	void removeCreature(std::shared_ptr<Creature> creature);
 
-	int32_t getThingIndex(const Thing* thing) const override final;
+	int32_t getThingIndex(std::shared_ptr<const Thing> thing) const override final;
 	size_t getFirstIndex() const override final;
 	size_t getLastIndex() const override final;
 	uint32_t getItemTypeCount(uint16_t itemId, int32_t subType = -1) const override final;
-	Thing* getThing(size_t index) const override final;
+	std::shared_ptr<Thing> getThing(size_t index) const override final;
 
-	void postAddNotification(Thing* thing, const Cylinder* oldParent, int32_t index,
+	void postAddNotification(std::shared_ptr<Thing> thing, std::shared_ptr<const Thing> oldParent, int32_t index,
 	                         cylinderlink_t link = LINK_OWNER) override final;
-	void postRemoveNotification(Thing* thing, const Cylinder* newParent, int32_t index,
+	void postRemoveNotification(std::shared_ptr<Thing> thing, std::shared_ptr<const Thing> newParent, int32_t index,
 	                            cylinderlink_t link = LINK_OWNER) override final;
 
-	void internalAddThing(Thing* thing) override final;
-	void internalAddThing(uint32_t index, Thing* thing) override;
+	void internalAddThing(std::shared_ptr<Thing> thing) override final;
+	void internalAddThing(uint32_t index, std::shared_ptr<Thing> thing) override;
 
-	const Position& getPosition() const override final { return tilePos; }
+	const Position& getPosition() const { return tilePos; }
```

Comment:
The override is missing here
```suggestion
	const Position& getPosition() const override { return tilePos; }
```